### PR TITLE
E0d/ad hoc reporting

### PR DESCRIPTION
**Pull request by e0d**
Re-factored role

### CODE REVIEW

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Dependencies:
+#   - common
+#
+- name: install system packages
+  apt: >
+    pkg={{ item }}
+    state=present
+  with_items: bastion_debian_pkgs
+
+- name: install bastion python packages
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: Nit: s/bastion/ad_hoc/

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  pip: >
+    name="{{ item }}" state=present
+    extra_args="-i {{ COMMON_PYPI_MIRROR_URL }}"
+  with_items: bastion_pip_pkgs
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: s/bastion_pip_pkgs/ad_hoc_reporting_pip_pkgs/  Looks like this update needs to be made in defaults as well.

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  with_items:
+    - /edx/bin
+  
+# These templates rely on there being a global
+# read_only mysql user, you must override the default
+# in order for these templates to be written out
+- name: install common mysql replica scripts
+  template: >
+    src=edx/bin/mysql.sh.j2
+    dest=/edx/bin//{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ item.script_name }}
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: nit: extra slash after `bin`

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  with_items:
+    - db_host: "{{ EDXAPP_MYSQL_REPLICA_HOST }}"
+      db_name: "{{ EDXAPP_MYSQL_DB_NAME }}"
+      script_name: edxapp-mysql.sh
+  when: COMMON_MYSQL_READ_ONLY_PASS is defined
+
+- name: install xqueue mysql replica scripts
+  template: >
+    src=edx/bin/mysql.sh.j2
+    dest=/edx/bin//{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ item.script_name }}
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: nit: extra slash after `bin`

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      script_name: xqueue-mysql.sh
+  when: COMMON_MYSQL_READ_ONLY_PASS is defined and XQUEUE_MYSQL_DB_HOST is defined
+  
+# These templates rely on there being a global
+# read_only mongo user, you must override the default
+# in order for these templates to be written out
+- name: install mongodb replica scripts
+  template: >
+    src=edx/bin/mongo.sh.j2
+    dest=/edx/bin//{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ item.script_name }}
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: nit: extra slash after `bin`

File: playbooks/roles/ad_hoc_reporting/defaults/main.yml [hunk 6]
```diff
@@ -18,15 +18,15 @@
...
 
 # These users are created on the bastion
 # server.
-BASTION_USER_INFO: []
+AD_HOC_REPORTING_USER_INFO: []
 #
 # vars are namespace with the module name.
 #
-bastion_role_name: bastion
+bastion_role_name: ad_hoc_reporting
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/defaults/main.yml]: s/bastion_role_name/ad_hoc_reporting_role_name/

Same with the lower case vars below.

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      script_name: edxapp-mysql.sh
+  when: COMMON_MYSQL_READ_ONLY_PASS is defined
+
+- name: install xqueue mysql replica scripts
+  template: >
+    src=edx/bin/mysql.sh.j2
+    dest=/edx/bin//{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-{{ item.script_name }}
+    mode=0755 owner=root group=root
+  with_items:
+    - db_host: "{{ XQUEUE_MYSQL_HOST }}"
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: It looks like you defined `XQUEUE_MYSQL_REPLICA_HOST` in the secrets so we should use that here.

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+      db_name: "{{ XQUEUE_MYSQL_DB_NAME }}"
+      script_name: xqueue-mysql.sh
+  when: COMMON_MYSQL_READ_ONLY_PASS is defined and XQUEUE_MYSQL_DB_HOST is defined
```
Comment (feanil) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: `XQUEUE_MYSQL_DB_HOST` should also be `XQUEUE_MYSQL_REPLICA_HOST`

File: playbooks/roles/ad_hoc_reporting/tasks/main.yml (hunk 3 above)
Comment (e0d) [inline playbooks/roles/ad_hoc_reporting/tasks/main.yml]: why?

**Comment by feanil** [comment]
:+1: :shipit:

**Pull request merged**